lib/segment/src/index/struct_payload_index.rs

```diff
@@ -217,14 +219,14 @@ impl StructPayloadIndex {
         Ok(field_indexes)
     }
 
-    fn build_and_save(&mut self, field: &PayloadKeyType) -> OperationResult<()> {
-        if !self.config.indexed_fields.contains(field) {
-            self.config.indexed_fields.push(field.clone());
+    fn build_and_save(&mut self, field: PayloadKeyTypeRef) -> OperationResult<()> {
+        if !self.config.indexed_fields.iter().any(|x| x == field) {
```

Comment:
I think `contains` is more readable in this case. Can you pls refer the explanation for this change.